src/components/GraphFilter/GraphFind.tsx

```diff
@@ -47,21 +51,28 @@ export class GraphFind extends React.PureComponent<GraphFindProps> {
       props.toggleFindHelp();
     }
 
-    this.hideInputRef = React.createRef();
-    this.hideInputValue = '';
-    this.hideValue = '';
-    this.hiddenElements = undefined;
-
     this.findInputRef = React.createRef();
     this.findInputValue = '';
```

Comment:
Do we really need this? This value should be taken from the prop value `findValue`, the same for the `hideValue` part.

Reply:
Also, i think you should use the prop (`findValue`) to initialize the `FormControl`
See `value` field 
```tsx
<FormControl
                type="text"
                value={this.props.findValue}
                style={{ width: '18em' }}
                inputRef={ref => {
                  this.findInputRef = ref;
                }}
                onChange={this.updateFind}
                onKeyPress={this.checkSubmitFind}
                placeholder="Find..."
              />
```

Reply:
> Do we really need this?

Are you asking whether we need this.findInputValue?   I think we do.  It is the current value being displayed in the input.  Without it the value typed by the user is not displayed, 
 Also, we don't want to update redux on every character typed.  This.props.findValue is different, it holds the last "submitted" value.

As for the line that actually initializes the variable to '', I don't think we actually need that line.

And I don't think the FormControl.value attribute works quite the way your expect.  Trying to use it causes some issues as it's not really treated just like an initial value.  It is better to set the ref value.

Reply:
Oh ok, so `this.findInputValue` is the current value of the input, not the value in effect.

Reply:
We still need for a way to initialize the  findInput based on the received prop.

I see that `componentDidUpdate` takes care of updating the value, but is not called in first render.

> componentDidUpdate() is invoked immediately after updating occurs. **This method is not called for the initial render**.

https://reactjs.org/docs/react-component.html#componentdidupdate

Reply:
OK, I pushed a change to add ComponentDidMount logic.  This is called after the first render and therefore the ref is set and can be manipulated.  Even if we found a way to use FormControl.value or FormControl.defaultValue to set the initial value, it would not be sufficient.  We would still need to perform the "submit" for the initial value, if set in redux at mount time.  What do you think?  I'm open to alternatives.